### PR TITLE
Dynamic point sampler

### LINKED ISSUE #12934: Expand PointSamplerBase a bit
## Rationale

PointSamplerBase has a lot of great capabilities, but only allows for a static sample size. I'd like to take advantage of the code in this class to work with a dynamically changing "mesh" from another application. 

## Description

In order to have a dynamic API, we just need to be able to set and call functions to work with a changing `_points` vector.

## Impact

Minor

### PR DESCRIPTION
**Pull request by permcody**
closes #12934

### CODE REVIEW

**Comment by permcody** [comment]
@aeslaughter - Why did I fail test coverage in the "test app"? I thought we weren't documenting test classes. The error seems to be telling me that my new test object needs a .md file in the test/doc directory.

https://civet.inl.gov/job/295036/

File: framework/src/vectorpostprocessors/PointSamplerBase.C [hunk 1]
```diff
@@ -144,6 +144,18 @@ PointSamplerBase::finalize()
   SamplerBase::finalize();
 }
 
+void
+PointSamplerBase::setPointsVector(std::vector<Point> points)
```
Comment (lindsayad) [inline framework/src/vectorpostprocessors/PointSamplerBase.C]: Is there a good reason why we're making a copy here?

File: test/tests/vectorpostprocessors/dynamic_point_sampler/tests [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  [dynamic_point_sampler_transfer]
+    type = CSVDiff
+    input = dynamic_point_sampler.i
+    csvdiff = 'dynamic_point_sampler_out_dynamic_line_sampler_0001.csv dynamic_point_sampler_out_dynamic_line_sampler_0006.csv'
+    cli_args = 'VectorPostprocessors/dynamic_line_sampler/test_transfer_points_vector=true'
+    prereq = 'dynamic_point_sampler'
+
+    issues = "#12934"
+    design = "LineValueSampler.md"
+    requirement = "The system shall support dynamic numbers of sample points during the simulation."
```
Comment (lindsayad) [inline test/tests/vectorpostprocessors/dynamic_point_sampler/tests]: Shouldn't we have a different requirement for a different test?

File: framework/src/vectorpostprocessors/PointSamplerBase.C (hunk 1 above)
Comment (permcody) [reply framework/src/vectorpostprocessors/PointSamplerBase.C]: Nope - but it doesn't really matter, the compiler should optimize one of these copies out. It won't copy twice when it makes another copy during the _points assignment. Note that I created a "transfer" version of this function which allows us to move a vector instead. I can fix this though just to be more clear.

File: test/tests/vectorpostprocessors/dynamic_point_sampler/tests (hunk 2 above)
Comment (permcody) [reply test/tests/vectorpostprocessors/dynamic_point_sampler/tests]: Yes

**Review by lindsayad — APPROVED** [review]

**Pull request merged**